File: Python/BOJ/10845.py
import sys
n = int(sys.stdin.readline().rstrip())

queue = []
for _ in range(n):
  command = list(sys.stdin.readline().rstrip().split())
  if command[0] == "push":
    queue.append(command[1])
  elif command[0] == "pop":
    if queue:
      print(queue[0])
      queue.remove(queue[0])
    else:
      print("-1")
  elif command[0] == "size":
    print(len(queue))
  elif command[0] == "empty":
    if queue: print("0")
    else: print("1")
  elif command[0] == "front":
    if queue: print(queue[0])
    else: print("-1")
  elif command[0] == "back":
    if queue: print(queue[-1])
    else: print("-1")
    

File: Swift/BOJ/10000/10845.swift
//
//  10845.swift
//  Algorithm
//
//  Created by 유정주 on 2023/08/10.
//

import Foundation

let n = Int(readLine()!)!
var queue: [String] = []

for _ in 0..<n {
    let input = readLine()!.split { $0 == " " }.map { String($0) }
    switch input[0] {
    case "push":
        queue.append(input[1])
    case "pop":
        if !queue.isEmpty {
            let item = queue.removeFirst()
            print(item)
        } else {
            print("-1")
        }
    case "size":
        print(queue.count)
    case "empty":
        print(queue.isEmpty ? "1" : "0")
    case "front":
        print(queue.first ?? -1)
    case "back":
        print(queue.last ?? -1)
    default: break
    }
}
